stactools_sentinel2/stactools/sentinel2/product_metadata.py

```diff
@@ -110,51 +142,35 @@ def metadata_dict(self):
             's2:product_uri':
             self.product_id,
             's2:generation_time':
-            get_xml_node_text(self.product_info_node, 'GENERATION_TIME'),
+            self.product_info_node.find_text('GENERATION_TIME'),
             's2:processing_baseline':
-            get_xml_node_text(self.product_info_node, 'PROCESSING_BASELINE'),
+            self.product_info_node.find_text('PROCESSING_BASELINE'),
             's2:product_type':
-            get_xml_node_text(self.product_info_node, 'PRODUCT_TYPE'),
+            self.product_info_node.find_text('PRODUCT_TYPE'),
             's2:datatake_id':
-            get_xml_node_attr(self.datatake_node, 'datatakeIdentifier'),
+            self.datatake_node.find_text('datatakeIdentifier'),
```

Comment:
The datatakeIdentifier is an attribute of the datatake_node, thus the `get_attr` method might be used instead of `find_text`.
`<Datatake datatakeIdentifier="GS2A_20200309T082751_024614_N02.14">`

Reply:
Great catch! Thank you, fixing and adding a test to cover this.